astrocut/make_cube.py

```diff
@@ -346,3 +349,500 @@ def make_cube(self, file_list, cube_file="img-cube.fits", sector=None, max_memor
+        self.primary_header = None
+        self.info_table = None
+        self.cube_file = None
+
+        # Used when updating an existing cube 
+        self.old_cols = None
+        self.update = False
+        self.cube_append = None
+
+    def _configure_cube(self, file_list, **extra_keywords):
+        """ Run through all the files and set up the  basic parameters for the cube.
+        Set up the cube primary header.
+        """
+
+        file_list = np.array(file_list)
+        image_shape = None
+        start_times = np.zeros(len(file_list))
+        for i, ffi in enumerate(file_list):
+            ffi_data = fits.open(ffi, mode='denywrite', memmap=True)
+            
+            start_times[i] = ffi_data[0].header.get(self.time_keyword)
+
+            if image_shape is None:  # Only need to fill this once
+                image_shape = ffi_data[0].data.shape
+            
+            if self.template_file is None:  # Only check this if we don't already have it
+
+                is_template = True
+                for key, value in self.template_requirements.items():
+                    if ffi_data[0].header.get(key) != value:  # Checking for a good image header
+                        is_template = False
+                        
+                if is_template:
+                    self.template_file = ffi
+                    
+            ffi_data.close()
+
+        self.file_list = file_list[np.argsort(start_times)]
+
+        # Working out the block size and number of blocks needed for writing the cube
+        # without using too much memory
+        slice_size = image_shape[1] * len(self.file_list) * 2 * 4  # in bytes (float32)
+        max_block_size = int((self.max_memory * 1e9)//slice_size)
+        
+        self.num_blocks = int(image_shape[0]/max_block_size + 1)
+        self.block_size = int(image_shape[0]/self.num_blocks + 1)
+        
+        # Determining cube shape: 
+        # If it's a new cube, the shape is (nRows, nCols, nImages, 2)
+        if not self.update:
+            self.cube_shape = (image_shape[0], image_shape[1], len(self.file_list), 2)
+            
+        # Else, if it's an update to an existing cube, the shape is (nRows, nCols, nImages + nNewImages, 2)
+        else: 
+            self.cube_shape = self.cube_append.shape
+            
+        # Making the primary header if there's no cube_file yet
+        if not self.update:
+            with fits.open(self.file_list[0], mode='denywrite', memmap=True) as first_file:
+                header = deepcopy(first_file[0].header)
+                header.remove('CHECKSUM', ignore_missing=True)
+
+                # Adding standard keywords
+                header['ORIGIN'] = 'STScI/MAST'
+                header['DATE'] = str(date.today())
+
+                # Adding factory specific keywords
+                for kwd in self.image_header_keywords:
+                    header[kwd] = (first_file[0].header[kwd], first_file[0].header.comments[kwd])
```

Comment:
can you add a comment here about why this is `first_file[0]` instead of `first_file[1]` assuming this is intended?